lib/src/barback/dartdevc/dartdevc_module_transformer.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:barback/barback.dart';
+import 'package:bazel_worker/bazel_worker.dart';
+import 'package:path/path.dart' as p;
+
+import '../../io.dart';
+import 'workers.dart';
+import 'module.dart';
+import 'module_reader.dart';
+import 'temp_environment.dart';
+import 'linked_summary_transformer.dart';
+
+/// Creates dartdevc modules given [moduleConfigName] files which describe a set
+/// of [Module]s.
+///
+/// Linked summaries should have already been created for all modules and will
+/// be read in to create the modules.
+class DartDevcModuleTransformer extends Transformer {
+  @override
+  String get allowedExtensions => moduleConfigName;
+
+  DartDevcModuleTransformer();
+
+  @override
+  Future apply(Transform transform) async {
+    TempEnvironment tempEnv;
```

Comment:
It looks like there's a lot of similarity in this code and [this stuff](https://github.com/dart-lang/pub/pull/1579/files#diff-76378714c8a4b4f95e6ef458e6269a20R36). Can it be refactored to reuse some of it? Maybe move more logic into Module itself?